diff --git a/setup.py b/setup.py
index d8213ce..651af14 100644
--- a/setup.py
+++ b/setup.py
@@ -7,4 +7,16 @@ from setuptools.command.build import build
 
 build.sub_commands.append(('build_idl', None))
 
-setuptools.setup()
+import os
+def find_data_files(src_dir, dest_dir):
+    data_files = []
+    for root, dirs, files in os.walk(src_dir):
+        install_path = os.path.join(dest_dir, os.path.relpath(root, src_dir))
+        file_paths = [os.path.join(root, file) for file in files]
+        data_files.append((install_path, file_paths))
+    return data_files
+example_data_files = find_data_files('OpenRTM_aist/examples', 'share/openrtm-2.0/components/python3')
+
+setuptools.setup(
+    data_files=example_data_files,
+)diff --git a/setup.py b/setup.py
